Index: qsnapshot-0.1.0/CMakeLists.txt
===================================================================
--- qsnapshot-0.1.0.orig/CMakeLists.txt	2012-04-14 20:32:43.781946239 +0800
+++ qsnapshot-0.1.0/CMakeLists.txt	2012-04-26 05:04:30.704165461 +0800
@@ -114,7 +114,9 @@
 install(TARGETS QSnapshot
     RUNTIME DESTINATION "bin" COMPONENT Runtime
     BUNDLE DESTINATION "." COMPONENT Runtime)
-install(SCRIPT "install.cmake" COMPONENT Runtime)
+if(NOT UNIX)
+    install(SCRIPT "install.cmake" COMPONENT Runtime)
+endif()
 
 # package part
 set(CPACK_PACKAGE_VERSION_MAJOR ${QSNAPSHOT_VERSION_MAJOR})
