--- a/src/widget/regiongrabber.cpp
+++ b/src/widget/regiongrabber.cpp
@@ -139,9 +139,9 @@
 	this->host->hide();
 	this->host->releaseMouse();
 	this->host->releaseKeyboard();
-	std::for_each( std::begin( this->handles ), std::end( this->handles ), []( QRect * r )->void {
-		r->moveTo( 0, 0 );
-	} );
+	for( auto it = this->handles.begin(); it != this->handles.end(); ++it ) {
+		( *it )->moveTo( 0, 0 );
+	}
 	this->selection = QRect();
 }
 
